tests/client_test.go

```diff
@@ -489,6 +489,17 @@ func TestStats(t *testing.T) {
 	if err != nil {
 		t.Errorf("Couldn't remove servers: %v", err)
 	}
+
+	locationZones := stats.LocationZones
+	if locationZones == nil {
+		t.Errorf("location zones are not available")
+	}
+
+	resolvers := stats.Resolvers
```

Comment:
Similar here. You'll have to configure a resolver to populate stats here. A `nil` check doesn't validate working stats.

Reply:
You can at least check that we have the right number of resolvers returned. You can start collecting resolvers stats by extending the existing resolver defined in `nginx.conf`:
```
resolver 127.0.0.11 valid=5s status_zone=test-zone;
```

Reply:
This test should happen before `_, _, err = c.UpdateHTTPServers(upstream, []client.UpstreamServer{})` just above. 

The upstream cleanup/deletion should always be the last step of the function.